lib/rules/no-arrow-functions-in-watch.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * @author Sosuke Suzuki
+ */
+'use strict'
+
+const utils = require('../utils')
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow using arrow functions to define watcher',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/no-arrow-functions-in-watch.html'
+    },
+    fixable: null,
+    schema: []
+  },
+  create (context) {
+    return utils.executeOnVue(context, (obj) => {
+      const watchNode = obj.properties.find((property) => utils.getStaticPropertyName(property) === 'watch')
+      if (watchNode == null) {
+        return
+      }
+      const watchValue = watchNode.value
+      if (watchValue.type !== 'ObjectExpression') {
+        return
+      }
+      for (const property of watchValue.properties) {
+        if (property.value.type === 'ArrowFunctionExpression') {
```

Comment:
Could you check `type` is a `"Property"`?
Probably get an error with "SpreadElement".

https://github.com/estree/estree/blob/master/es2015.md#expressions
https://github.com/estree/estree/blob/master/es2018.md#expressions